Funcky/Monads/Either/EitherExtensions.cs

```diff
@@ -3,16 +3,24 @@ namespace Funcky.Monads;
 public static class EitherExtensions
 {
     /// <summary>Returns the left value or <see cref="Option{TItem}.None()"/> if the <paramref name="either"/> is a right value.</summary>
+    [Pure]
     public static Option<TLeft> LeftOrNone<TLeft, TRight>(this Either<TLeft, TRight> either)
         where TLeft : notnull
         => either.Match(
             left: Option.Some,
             right: static _ => Option<TLeft>.None());
 
     /// <summary>Returns the right value or <see cref="Option{TItem}.None()"/> if the <paramref name="either"/> is a left value.</summary>
+    [Pure]
     public static Option<TRight> RightOrNone<TLeft, TRight>(this Either<TLeft, TRight> either)
         where TRight : notnull
         => either.Match(
             left: static _ => Option<TRight>.None(),
             right: Option.Some);
+
+    [Pure]
+    public static Either<TRight, TLeft> Flip<TRight, TLeft>(this Either<TLeft, TRight> either)
```

Comment:
Why is this an extension method and not a method on Either?

Reply:
Because it is not core functionality on Either, it's obviously an add-on since we implement it now.